src/v/cluster/tests/serialization_rt_test.cc

```diff
@@ -452,3 +453,78 @@ SEASTAR_THREAD_TEST_CASE(partition_status_serialization_old_version) {
         BOOST_CHECK(result[i].leader_id == original[i].leader_id);
     }
 }
+
+template<typename T>
+void roundtrip_test(const T original) {
```

Comment:
nit: `const ref` and concept for T as a type that is "serde-able"

Reply:
Something like this?
```suggestion
template<typename T>
concept SerdeCompatible = serde::is_serde_compatible_v<T>;

void roundtrip(const SerdeCompatible auto& original) {
```

Reply:
thats cool, we could put that concept in a header somewhere too

Reply:
> there is something similar in serde.h already

I think there's just the trait, it's devoid of concepts last I saw. Which makes sense based on the compiler we had when it was written.

Reply:
None of this is a blocker for me, just informational notes, if it helps.

Reply:
Thanks I'm going to have to consume this. For the moment this is a tiny helper in a test.

Reply:
@BenPope this is awesome, thanks for the write up. do you know what this `void roundtrip(const SerdeCompatible auto& original) { ... }` syntax is called?

Reply:
> @BenPope this is awesome, thanks for the write up. do you know what this `void roundtrip(const SerdeCompatible auto& original) { ... }` syntax is called?

Abbreviated function template.

Reply:
wild

Reply:
+1 this is very cool